sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -90,6 +90,15 @@ abstract class JdbcDialect extends Serializable {
   }
 
   /**
+   * Parses a table name in case of containing special characters like . seperating database name
+   * followed by table name (eg "some database"."some-table-name") or 
+   * in case it contains characters that require quotes (e.g. space).
+   */
+  def parseTableName(tableName: String): String = {
```

Comment:
The method contract here is confusing to me: is this supposed to introduce an additional layer of quotes? I find it confusing to couple both the responsibilities of quoting identifiers and parsing table names from a schema-qualified name.

I'd prefer to have this return just the table name, unquoted, and leave it up to later components to escape it. If you _do_ think we should do quoting here, please explain that part of the API contract in the Scaladoc.